### PR TITLE
Handle packed files

### PR DESCRIPTION
**Pull request by rod-glover**
This addresses #36

### CODE REVIEW

File: dp/generate_climos.py [hunk 1]
```diff
@@ -134,19 +134,34 @@ def climo_outputs(time_resolution):
     # Apply per-variable processing
     # - Scale pr variable if it is not in desired units
     #   - TODO: Use UDUNITS here for more robust processing of units? It has a fairly heavy API. But this is uuuugly.
```
Comment (jameshiebert) [inline dp/generate_climos.py]: You may want to check out the [``pint`` library](https://pint.readthedocs.io/en/0.7.2/) for units handling.

File: dp/generate_climos.py [hunk 2]
```diff
@@ -134,19 +134,34 @@ def climo_outputs(time_resolution):
...
+    pr_attributes = {}  # will contain updates, if any, to pr variable attributes
     if 'pr' in input_file.dependent_varnames:
```
Comment (jameshiebert) [inline dp/generate_climos.py]: I know that you didn't create this problem, but I don't like the big, long branch here for `pr` specific code. Could we stash that away in a subroutine or something to help this read better for the common cases?

File: dp/generate_climos.py [hunk 3]
```diff
@@ -134,19 +134,34 @@ def climo_outputs(time_resolution):
...
-            # Extract variable
-            pr_only = cdo.select('name=pr', input=climo_means)
-            # Multiply values by 86400 to convert from mm/s to mm/day
-            pr_only = cdo.mulc('86400', input=pr_only)
-            # Replace pr in all-variables file
-            climo_means = cdo.replace(input=[climo_means, pr_only])
             # Update units attribute
-            # TODO: Verify that "d-1" is desired way to express "per day" (alternaive is "day-1")
-            pr_units_attr = re.sub('(/s|/ s|s-1|s\^-1)', ' d-1', units)
+            pr_attributes['units'] = re.sub('(/s|/ s|s-1|s\^-1)', ' d-1', units)
```
Comment (jameshiebert) [inline dp/generate_climos.py]: I've seen s-1 for "per second", but I've never seen d-1 for "per day". Is that in the SI units?

File: dp/generate_climos.py [hunk 4]
```diff
@@ -169,11 +184,9 @@ def climo_outputs(time_resolution):
         cf.climo_start_time = t_start.isoformat()[:19] + 'Z'
         cf.climo_end_time = t_end.isoformat()[:19] + 'Z'
         if hasattr(input_file, 'tracking_id'):
-            cf.climo_tracking_id=input_file.tracking_id
-        try:
-            cf.variables['pr'].units = pr_units_attr
-        except NameError:
```
Comment (jameshiebert) [inline dp/generate_climos.py]: Silent errors are definitely bad, so I approve of getting rid of this. What actually happens here if the variable is not attributed with units? I guess we're assuming that everything going through this script passes nclint and therefore *is* attributed with units, eh? I'm fine with that, just making sure.

File: dp/generate_climos.py (hunk 1 above)
Comment (rod-glover) [inline dp/generate_climos.py]: What a delightful and elegant library! `pint` can't parse all valid `udunits` strings, but a simple transformation of the strings can fix that. 

Also the Python binding of `udunits2`, `cfunits`, only works in Python ver >=2.6 and <3.0, so we **must** use something different. I'm liking `pint`.

File: dp/generate_climos.py (hunk 3 above)
Comment (rod-glover) [inline dp/generate_climos.py]: 'd' (and for that matter 'day') is not an SI unit. But 'day' commonly used in extended units sets, such as those accepted by `udunits2` and `pint`. 'd' is recognized as a synonym for 'day' by `udunits2` (but not `pint`). 'd' is commonly used by scientists at PCIC.

File: dp/generate_climos.py (hunk 1 above)
Comment (rod-glover) [inline dp/generate_climos.py]: There is another Python binding of udunits2 to Python,[ `cf_units`](https://github.com/SciTools/cf_units). It appears to be Python 3 compatible. It's written by the UK Met office, and appears to be actively maintained. Upside: Exactly what is called for in this situation. Downside: Have to install `udunits2` separately.

File: dp/generate_climos.py (hunk 4 above)
Comment (rod-glover) [inline dp/generate_climos.py]: An answer in two parts:

1. The original try-catch block was just a shitty way decide if there were any updates to be done. Much better now.
2. If the variable is not actually attributed with units, the code will raise an AttributeError exception from netcdf. But we don't yet have an nclint check that all variables are attributed with units. (And we could probably add some other attributes to check as well.)

**Comment by jameshiebert** [comment]
To me, this all looks good to go (except for the one line conflict).

**Pull request merged**